src/commands/flutter-symbols/upload.ts

```diff
@@ -0,0 +1,444 @@
+        'flutter:true',
+        'platform:android',
+      ],
+      prefix: 'datadog.ci.symbols.upload.',
+    })
+
+    const apiKeyValidator = newApiKeyValidator({
+      apiKey: this.config.apiKey,
+      datadogSite: this.config.datadogSite,
+      metricsLogger: metricsLogger.logger,
+    })
+
+    const initialTime = Date.now()
+
+    const callResults: UploadStatus[] = []
+    try {
+      if (this.iosDsymsLocation) {
+        callResults.push(await this.performDsymUpload())
+      }
+      if (this.androidMappingLocation) {
+        callResults.push(await this.performAndroidMappingUpload(metricsLogger, apiKeyValidator))
+      }
+      if (this.dartSymbolsLocation) {
+        callResults.push(...(await this.performDartSymbolsUpload(metricsLogger, apiKeyValidator)))
+      }
+
+      const totalTime = (Date.now() - initialTime) / 1000
+      this.context.stdout.write(renderCommandSummary(callResults, totalTime, this.dryRun))
+    } catch (e) {
+      this.context.stderr.write(renderGeneralizedError(e))
+
+      return 1
+    } finally {
+      try {
+        await metricsLogger.flush()
+      } catch (err) {
+        this.context.stdout.write(`WARN: ${err}\n`)
+      }
+    }
+
+    return 0
+  }
+
+  private getAndroidMetadata(): MappingMetadata {
+    return this.getMappingMetadata(TYPE_JVM_MAPPING)
+  }
+
+  private getFlutterMetadata(platform: string, arch: string) {
+    return this.getMappingMetadata(TYPE_DART_SYMBOLS, platform, arch)
+  }
+
+  private getFlutterSymbolFiles(dartSymbolLocation: string): string[] {
+    const symbolPaths = glob.sync(buildPath(dartSymbolLocation, '*.symbols'))
+
+    return symbolPaths
+  }
+
+  private getGitDataPayload(): MultipartValue | undefined {
+    if (this.gitData === undefined) {
+      return undefined
+    }
+
+    const files = this.gitData.trackedFilesMatcher.rawTrackedFilesList()
+    const repoPayload = {
+      data: [
+        {
+          files,
+          hash: this.gitData.hash,
+          repository_url: this.gitData.remote,
+        },
+      ],
+      version: 1,
+    }
+
+    return {
+      options: {filename: 'repository', contentType: 'application/json'},
+      value: JSON.stringify(repoPayload),
+    }
+  }
+
+  private async getGitMetadata(): Promise<RepositoryData | undefined> {
+    try {
+      return await getRepositoryData(await newSimpleGit(), this.repositoryUrl)
+    } catch (e) {
+      this.context.stdout.write(renderGitWarning(e))
+    }
+
+    return undefined
+  }
+
+  private getMappingMetadata(type: string, platform?: string, arch?: string): MappingMetadata {
+    return {
+      arch,
+      cli_version: this.cliVersion,
+      git_commit_sha: this.gitData?.hash,
+      git_repository_url: this.gitData?.remote,
+      platform,
+      service: this.serviceName,
+      type,
+      variant: this.flavor,
+      version: this.version!,
+    }
+  }
+
+  private async parsePubspec(pubspecLocation: string): Promise<number> {
```

Comment:
Minor comment: this function's goal is to read the version from the pubspec and set the `version` of the command, this could be reflected in its name, like `parseVersionFromPubspec`. If you'd like to keep this one's name more generic, maybe this can also be achieved in splitting it into a part that parses the pubspec, and one part that reads and sets the version.